third_party/terraform/resources/resource_cloudfunctions_function.go

```diff
@@ -275,6 +275,13 @@ func resourceCloudFunctionsFunction() *schema.Resource {
 				Removed:  "This field is removed. Use `event_trigger.failure_policy.retry` instead.",
 			},
 
+			"max_instances": {
+				Type:             schema.TypeInt,
+				Optional:         true,
+				ValidateFunc:     validation.IntAtLeast(0),
+				DiffSuppressFunc: emptyOrDefaultStringSuppress("0"),
```

Comment:
Isn't the default `0`? And isn't this an int, not a string?

Reply:
Technically if default is 0 it'll always send 0 vs not sending it? But they act the same and eventually it'll be 0 so it's nbd, I'll change it. It'll let people unset it that way (when they remove it). As for DSF, diffs are done on the string values from state/config (function signature is `func(k, old, new string, ...)`. Default diff does per-type comparison (but still uses diffString)